### DISCUSSION TITLE
write error: Writing characteristic failed with status code 200

### DISCUSSION THREAD

**Discussion by brandomcombr** [capacitor-community/bluetooth-le#616] [Q&A] [answered]
Hi Community, 

When I call the write method, I'm getting the error: Writing characteristic failed with status code 200

I saw that in the Device.kt file, there is this condition.
```
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            val statusCode = bluetoothGatt?.writeCharacteristic(characteristic, bytes, writeType)
            if (statusCode != BluetoothStatusCodes.SUCCESS) {
                reject(key, "Writing characteristic failed with status code $statusCode.")
                return
            }
        } else {
            characteristic.value = bytes
            characteristic.writeType = writeType
            val result = bluetoothGatt?.writeCharacteristic(characteristic)
            if (result != true) {
                reject(key, "Writing characteristic failed.")
                return
            }
        }
```

When I enter this first condition, do I need any additional permissions or need to call any other method besides connect to invoke write?

**Comment by pwespi** [comment, marked as answer]
Status code 200 means [ERROR_GATT_WRITE_NOT_ALLOWED](https://developer.android.com/reference/android/bluetooth/BluetoothStatusCodes#ERROR_GATT_WRITE_NOT_ALLOWED). This means the peripheral device does not allow writing to this characteristic. You can check the characteristic properties using [`getServices`](https://github.com/capacitor-community/bluetooth-le?tab=readme-ov-file#getservices)
